src/test/groovy/com/github/rodionovsasha/commentservice/services/TopicServiceTest.groovy

```diff
@@ -0,0 +1,233 @@
+    }
+
+    def "archive throws when user not found"() {
+        given:
+        !topicService.getById(TOPIC_ID).archived
+
+        when:
+        topicService.archive(TOPIC_ID, NOT_EXISTING_USER_ID)
+
+        then:
+        !topicService.getById(TOPIC_ID).archived
+
+        and:
+        thrown(UserNotFoundException)
+    }
+
+    def "archive throws when user is not topic owner"() {
+        given:
+        def TOPIC_ID = 3
+        !topicService.getById(TOPIC_ID).archived
+
+        when:
+        topicService.archive(TOPIC_ID, HOMER_ID)
+
+        then:
+        !topicService.getById(TOPIC_ID).archived
+
+        and:
+        def e = thrown(TopicAccessException)
+        e.message == "Non-topic owner with id '1' is trying to archive the topic"
+    }
+
+    def "archive does not change already archived topic"() {
+        given:
+        def TOPIC_ID = 7
+        topicService.getById(TOPIC_ID).archived
+
+        when:
+        topicService.archive(TOPIC_ID, 1)
+
+        then:
+        topicService.getById(TOPIC_ID).archived
+    }
+
+    def "listForUser returns all topics for user ASC sorted"() {
+        when:
+        def topics = topicService.listForUser(HOMER_ID, new Sort(Sort.Direction.ASC, "id"))
+
+        then:
+        topics.id == [1, 2, 5, 6, 7]
+    }
+
+    def "listForUser returns all topics for user DESC sorted"() {
+        when:
+        def topics = topicService.listForUser(HOMER_ID, new Sort(Sort.Direction.DESC, "id"))
+
+        then:
+        topics.id == [7, 6, 5, 2, 1]
+    }
+
+    def "listForUser returns an empty list when user does not have own topics"() {
+        when:
+        def topics = topicService.listForUser(3, new Sort(Sort.Direction.ASC, "id"))
+
+        then:
+        topics.isEmpty()
+    }
+
+    def "listForUser throws when user is inactive"() {
+        when:
+        topicService.listForUser(2, null)
+
+        then:
+        thrown(InactiveUserException)
+    }
+
+    def "listForUser throws when user not found"() {
+        when:
+        topicService.listForUser(NOT_EXISTING_USER_ID, null)
+
+        then:
+        thrown(UserNotFoundException)
+    }
+
+    def "search returns topics with a fragment in title"() {
+        when:
+        def topics = topicService.search("Flanders", 10)
+
+        then:
+        topics.id == [6, 1]
+    }
+
+    def "search returns topics with a fragment in title case insensitive"() {
+        when:
+        def topics = topicService.search("FLANDERS", 10)
+
+        then:
+        topics.id == [6, 1]
+    }
+
+    def "search returns topics with a fragment in title with sorting"() {
+        when:
+        def topics = topicService.search("Flanders", 10)
+
+        then:
+        topics.get(0).date > topics.get(1).date
+    }
+
+    def "search returns nothing"() {
+        when:
+        def topics = topicService.search("Springfield", 10)
+
+        then:
+        topics.isEmpty()
+    }
+
+    def "search returns limited result"() {
+        when:
```

Comment:
This test does not show that without the limit of 2 the `search` would return more results. Can we add a `given` section to check that?
```groovy
given:
topicService.search("e", 100).size() = 6 // or some other right number
```